### PR TITLE
chore: use execute methods & types

### PR DESCRIPTION
**Pull request by boneskull**
This PR:

1. Replaces the hand-rolled Execute Methods with the builtin execute method facilities
2. Typechecks the entire project, which is basically needed if we're going to get reasonable docs.
3. Cleans up various docstrings (since they will be actually used in the documentation!)

Each method corresponding to an Execute Method (I think these methods always have prefix `mobile`) has had its function signature changed.  Instead of passing in an "options object", each now accepts discrete arguments.  I don't believe this is ideal, and I _expect these changes to churn_ later once we revisit how execute methods work (they _should_ allow "options objects").

It's possible that--because some methods (perhaps unwisely) allow arbitrary data to be sent thru Appium into other endpoints (like wda)--that the changes to the signatures will break somebody.  We will need to address these on a case-by-case basis, because it's simply not safe to do this, and we need to be explicit about what data we pass through.  Consider this a vulnerability mitigation. 😜

There are a few helper methods I've written or rewrote to support both the signature change and type changes.  There was also at least one legit bugfix (see `gesture.js:L117`)!

This line appears often in the changes:

```js
// @ts-expect-error - do not assign arbitrary properties to `this.opts`
```

Before we started typing things, `this.opts` (or anything else really) was fair game for whatever arbitrary stuff a driver wanted to put anywhere.  That's no longer the case.  Everything that this (or any other) driver is putting on `this.opts` which is not already declared should _not_ live there.  A better place may just be to use the instance's own context; I don't see any real distinction between the data already stored in the context vs. the data added to `this.opts`.

Some types have been moved out of `@typedef`s and into TS sources.

### CODE REVIEW

**Comment by boneskull** [comment]
Current dependencies on/for this PR:
* master
  * **PR #1484** <a href="https://app.graphite.dev/github/pr/appium/appium-xcuitest-driver/1484" target="_blank">Graphite</a> 
    * **PR #1499** <a href="https://app.graphite.dev/github/pr/appium/appium-xcuitest-driver/1499" target="_blank">Graphite</a> 
      * **PR #1531** <a href="https://app.graphite.dev/github/pr/appium/appium-xcuitest-driver/1531" target="_blank">Graphite</a> 
      * **PR #1500** <a href="https://app.graphite.dev/github/pr/appium/appium-xcuitest-driver/1500" target="_blank">Graphite</a> 
        * **PR #1533** <a href="https://app.graphite.dev/github/pr/appium/appium-xcuitest-driver/1533" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/appium/appium-xcuitest-driver/1533?utm_source=stack-comment).

**Comment by boneskull** [comment]
Either this or `@appium/typedoc-plugin-appium` will need some further changes to get the execute method map documentation generating properly.  Might just be a visibility issue...

File: lib/commands/biometric.js
Comment (jlipps) [inline lib/commands/biometric.js:30]: Simulatory?

File: lib/commands/element.js
Comment (jlipps) [inline lib/commands/element.js:203]: typically we have followed the `return await` pattern as a style rule so it's clear just by looking (without needing a type checker) that we're dealing with an async function

File: lib/driver.js
Comment (jlipps) [inline lib/driver.js:1336]: the things in this list are capabilities, so they should be OK on `this.opts` (i.e., they're not arbitrary properties). probably they're missing from the cap constraints.

**Review by jlipps — APPROVED** [review]
made a few comments, otherwise LGTM when tests are all green.

File: lib/driver.js
Comment (boneskull) [reply lib/driver.js:1336]: hmm, which?

File: lib/commands/element.js
Comment (boneskull) [reply lib/commands/element.js:203]: nothin gets by you, huh?? 😄

a few reasons why:

1. `await someFunction()` does not imply that `someFunction()` returns a `Promise`. Maybe `someFunction()` _used_ to return a `Promise`, and now you have potentially an entire chain of function calls which are `async` that needn't be.
2. if we always must use `return await someFunction()` then we are effectively documenting the return value of that function twice--and because of 1, this may be incorrect. this is akin to "no comments are better than wrong comments"
3. it is insignificantly slower (but it's slower)

if you want to pull rank, I'll change it, but I really hope to convince you & the team that this is unnecessary at best and harmful at worst.

Comment (mykola-mokhnach) [reply lib/commands/element.js:203]: I also like to keep `await`. Readability is more important than a (very) minor performance gain

File: lib/driver.js
Comment (boneskull) [reply lib/driver.js:1336]: ehh maybe just this one. I'll add it

Comment (boneskull) [reply lib/driver.js:1336]: @jlipps OK, so I'm looking at this, and `forceSimulatorSoftwareKeyboardPresence` is mentioned in the capability docs.

`shouldUseCompactRepsonses` is mentioned in the _settings_ documentation, as is `elementResponseAttributes` (both in this data structure).

Should these _all_ be added as constraints?  I also note that _`elementResponseFields`_ is referenced several times in the codebase (instead of +`elementResponseAttributes`_), which may be a bug?

cc @mykola-mokhnach @KazuCocoa

Comment (KazuCocoa) [reply lib/driver.js:1336]: For `elementResponseFields`. It is maybe no longer used since it is not in appium/WDA as well. It is in unit tests and capabilities for WDA, but no places use the value

Comment (KazuCocoa) [reply lib/driver.js:1336]: An item that was addressed only in settings can be set only via `settings[<settings name>]` as capabilities for now, as a documented way. 

Some can be set via capabilities as https://github.com/appium/WebDriverAgent/blob/master/WebDriverAgentLib/Commands/FBSessionCommands.m#L101-L123

Maybe we should print deprecated messages in xcuitest about capabilities that are available as capabilities (as implementation) for WDA, but not in capabilities docs. Then remove them from capabilities completely.

File: lib/commands/element.js
Comment (jlipps) [reply lib/commands/element.js:204]: if we were making this rule today i would agree with you given the extra layer of type safety that we now have to make clear what is a promise. at this point i'm purely concerned about inconsistency leading to confusion when reading code.

if you want to open another PR where you `s/return await/return/g` the entire codebase, i'm open to considering that, assuming other maintainers are OK with it as well.

Comment (boneskull) [reply lib/commands/element.js:204]: @mykola-mokhnach What of my first two points above?  It's not "good readability" if it's misleading.

Comment (mykola-mokhnach) [reply lib/commands/element.js]: I would prefer to have an async function that is synchronous "internally" rather than async one that is never awaited properly. `await syncFunc()` is totally fine in JS and does not create any issues while `asyncFunc()` (e.g. `await` is missing) might  create serious problems in the code flow which are hard to track down.

Comment (boneskull) [reply lib/commands/element.js]: @mykola-mokhnach Given the state of our type support, we can be sure at this point that a missing `await` would fail the build.

**Pull request merged**
